blockchain/contracts/HelloWorld.sol

```diff
@@ -66,16 +90,20 @@ contract HelloWorld {
             nftOwnerships[availableNFTs[0]].dropTime <= block.timestamp,
             "Droptime not yet reached!"
         );
-        uint256 i;
         uint256 nftElement;
         uint256 nftHash;
-        for (i = 0; i < joinedUsers.size(); i++) {
-            nftElement = generateRandomNumber(i) % availableNFTs.length;
-            nftHash = availableNFTs[nftElement];
-            remove(nftElement);
-            nftOwnerships[nftHash].reservedFor = joinedUsers.getAddressAtIndex(
-                i
-            );
+        for (uint256 i = 0; i < joinedUsers.size(); i++) {
+            for (
+                uint256 j = 0;
+                j < nftReservations[joinedUsers.getAddressAtIndex(i)];
+                j++
+            ) {
+                nftElement = generateRandomNumber(j) % availableNFTs.length;
+                nftHash = availableNFTs[nftElement];
+                remove(nftElement);
+                nftOwnerships[nftHash].reservedFor = joinedUsers
+                    .getAddressAtIndex(i);
```

Comment:
What happens if two contracts are executed at nearly the same time?
If only some or one is left, there is a certain probability.
Raises remove an error if it is not possible?

Reply:
I misunderstood the operating principle of the solidity techstack - this case is negligible.